debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresStreamingChangeEventSource.java

```diff
@@ -439,6 +439,7 @@ public void commitOffset(Map<String, ?> partition, Map<String, ?> offset) {
             }
         }
         catch (SQLException e) {
+            errorHandler.setProducerThrowable(e);
```

Comment:
@brewneaux Is this necessary here? As the exception is rethrown it should be caught by the change in `ChangeEventSourceCoordinator`. The `errorHandler` object is a single one per connector so it will be set there.

Reply:
Also IMHO there should be the same cleanup as is in the `execute` method so it should be extracted there to a separate method. But at the same time I believe it is possible that the calls are done from separate threads. So maybe there should be a flag field that is set in `commitOffset` and the `execute` method will fail when it is set. This might be the safest approach.

Reply:
In the testing I did on my initial PR, it appeared like it was. I just retested, though, in a few other scenarios with just the `ChangeEventSourceCoordinator` change and it continues to work as desired.

Reply:
@jpechane - just updated this PR. I did my best to try to understand what you were asking for and make it work within this context.